### ISSUE TITLE
Rule proposal: don't log when re-throwing a caught error

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#1537]
I don't know if this applies everywhere, but it feels that if you're catching an error just to log it and then throwing it again, you're potentially passing the error to another step that will do the same, resulting in the same error being logged twice.

Alternatives:

- only log the error in the last step, e.g. `log('there was an error', error); exit(1)`
- if your intent is to add more details, create a new Error + `Error#cause`

## Fail

```js
try {
	null();
} catch (error) {
	console.log('Error while doing stuff', error); // Log
	throw error; // and re-throw
}
```

## Pass

```js
try {
	null();
} catch (error) {
	console.log(error); // Only log
}
```

```js
try {
	null();
} catch (error) {
	doSomething(error);
	throw error; // Re-throw
}
```

```js
try {
	null();
} catch (error) {
	const specificError = new Error('Error while doing stuff')
	specificError.cause = error
	throw specificError; // Throw new error with more details
}
```

**Comment by sindresorhus** [comment]
Maybe this could be part of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1342?

**Comment by fregante** [comment]
> Maybe this could be part of #1342?

Potentially, yes, since the solution is:

- create new error to add more info + use cause; or
- just drop the log

**Comment by sindresorhus** [comment]
Reviewed this again; I dont think we should add this to Unicorn.

The related “throwing a new error without preserving the caught one” case is now covered by ESLint core’s `preserve-caught-error` rule, which is the clearer correctness issue.

The remaining case here, logging and then rethrowing the same error, is more of an application logging policy. It can be noisy, but it can also be intentional for telemetry, debugging, or a boundary where the caller will not log. A lint rule would either be too narrow to be useful or grow options and heuristics around loggers nd control flow.
